scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/InProcessTaskExecutor.java

```diff
@@ -209,10 +214,26 @@ private Serializable execute(TaskContext taskContext, PrintStream output, PrintS
         if (taskContext.getPreScript() != null) {
             Script<?> script = taskContext.getPreScript();
             forkedTaskVariablesManager.replaceScriptParameters(script, thirdPartyCredentials, variables, error);
-            ScriptResult preScriptResult = scriptHandler.handle(script, output, error);
-            if (preScriptResult.errorOccured()) {
-                throw new TaskException("Failed to execute pre script: " + preScriptResult.getException().getMessage(),
-                                        preScriptResult.getException());
+            Map<String, String> genericInfo = taskContext.getInitializer().getGenericInformation();
+            if (genericInfo != null && genericInfo.containsKey("PRE_SCRIPT_AS_FILE")) {
+                String path = genericInfo.get("PRE_SCRIPT_AS_FILE");
+                ScriptEngineFactory factory = new ScriptEngineManager().getEngineByName(script.getEngineName())
+                                                                       .getFactory();
+                String extension = factory.getExtensions().get(0);
```

Comment:
I think this could be made a bit more robust:
 - if the user does not provide an extension (path does not contain "."), you can eventually append factory.getExtensions().get(0)
 - otherwise, use path unchanged